### ISSUE TITLE
Get a hashtree of records through an association?

### ISSUE THREAD

**Issue by diegoviola** [mceachen/closure_tree#171]
So I have an association like this:

```ruby
class Post < ActiveRecord::Base
  has_many :comments
end

class Comment < ActiveRecord::Base
  belongs_to :post
  acts_as_tree order: 'created_at DESC'
end
```

What I'd like to do is to get a hashtree of all my comments through `@post`. E.g.

```ruby
@post = Post.find(1)
@comments = @post.comments.hash_tree
```

Is there a way to do this? When I try that I get records but not as a hash tree.

**Comment by adamjgrant** [comment]
Also having this problem.

If I do a `hash_tree` on `Comment` it works, of course, but gives me all comments ever.

I also experience this problem when returning other ActiveRecord relations

    comments = Comment.where(opinion_id: params[:opinion_id])
    @comment_tree = comments.hash_table

**Comment by adamjgrant** [comment]
My guess on this would be that `.hash_table` expects a single parent object.  Because `@post.comments` is an array, the function doesn't return appropriately.

One could instead do

    @post.hash_tree

But this would require structuring the `@post` itself as a table. This doesn't seem like the right way to go.

**Comment by adamjgrant** [comment]
For the time being, I was able to get it to work as such:

    comments = Opinion.find(params[:opinion_id]).comments
    @comments = Array.new

    comments.each do |comment|
      @comments.push Comment.json_tree(comment.hash_tree).first
    end

    respond_to do |format|
      format.json { render :json => @comments, root: false }
    end

You may not need the `json_tree` method unless you're outputting json. If so, here's my implementation in Comment.rb

    def self.json_tree(nodes)
      nodes.map do |node, sub_nodes|
        {:opinion_id => node.opinion_id, :body => node.body, :id => node.id, :parent_id => node.parent_id, :comments => Comment.json_tree(sub_nodes).compact}
      end
    end

**Comment by mceachen** [comment]
The new version of `closure_tree` that I just released, v5.2.0, adds support for `.children.hash_tree`.

For other collections, for example, for your `@post.comments.hash_tree` example, you'll need to do the following:

```ruby
@post.comments.map(&:hash_tree).reduce(:merge)
```
